### PR TITLE
fix: Incorrect session details display from widget list item click

### LINKED ISSUE #1825: Incorrect redirect to session details on clicking a list item in the widget
**Actual Behaviour**

If the app is offline and the user clicks a different bookmarked session from the bookmark widget, he gets directed to the same session details of the first clicked list item.

**Expected Behaviour**

The user should be directed to the corresponding session details activity of the corresponding clicked list item.

**Steps to reproduce it**

1. Get the bookmark widget.
2. Close the app.
3. Click the first item on the widget list view.
4. Press the home button.
5. Click any other item in the list view of the widget.

**Would you like to work on the issue?**

Yes.

**Target Milestone to solve the issue**

[milestone 11](https://github.com/fossasia/open-event-android/milestone/34)

### PR DESCRIPTION
**Pull request by rohanagarwal94**
Fixes #1825

Changes:
Also fixes the multiple stacking of activities when clicked on multiple list items.

@iamareebjamal @Shailesh351 @arp95 @shreyasnbhat @mahikaw please review.

### CODE REVIEW

File: android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java [hunk 1]
```diff
@@ -435,6 +435,10 @@ protected void onResume() {
 
         final SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
 
+        id = getIntent().getIntExtra(ConstantStrings.ID, 0);
```
Comment (Shailesh351) [inline android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java:438]: Getting this value from Extras creating problem in onCreate() method?
We are already  getting id in the onCreate() method. 
https://github.com/rohanagarwal94/open-event-android/blob/ec9f4ab95d166fddcd31b559b61b6adc376079ac/android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java#L145

Comment (rohanagarwal94) [reply android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java:438]: The problem is when the app is stopped, so does the lifecycle. Therefore when a list item is clicked, the onCreate method is never called and therefore the sessiondetailsactivity always showed the first clicked item details on any subsequent clicks. So it was necessary to fetch them from the intent in the onresume itself.

**Review by Shailesh351 — APPROVED** [review]

**Review by iamareebjamal — APPROVED** [review]
This is a good fix, but we need to remove the duplicate code in these classes. There shouldn't be same method calls in onCreate and onResume. Probably move everything in onResume. 

Probably in future

**Comment by shreyasnbhat** [comment]
This was a neat fix. Nice work.

_Sent from my OnePlus ONEPLUS A5000 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Pull request merged**